### PR TITLE
feature: Github CI integration

### PR DESCRIPTION
**Pull request by adam-murray**

### CODE REVIEW

**Review by michaeljcollinsuk — CHANGES_REQUESTED** [review]

File: tests/requirements/requirements_base.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+asgiref
+coverage>=4.4.2
+django-app-helper
+flake8>=3.0.4
+tox>=2.9.1
+
```
[the remark this answers is not in the document]
Comment (adam-murray) [reply tests/requirements/requirements_base.txt:7]: Good call, have added.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+on: [push, pull_request]
+
+jobs:
+  unit-tests:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [ 3.7, 3.8, 3.9 ]  # latest release minus two
```
Comment (Bernardvdv) [inline .github/workflows/test.yml:11]: Do we still need 3.7?

File: tests/requirements/requirements_base.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (michaeljcollinsuk) [reply tests/requirements/requirements_base.txt:7]: The test requirements should include everything required to run tests without having to also install from setup.py, without the other core requirements here you still get errors when installing only from e.g. `tests/requirements/dj32_cms40.txt`:
```
/Users/michael.collins/code/eliga/aldryn-redirects/venv/bin/python /Users/michael.collins/code/eliga/aldryn-redirects/tests/settings.py
Traceback (most recent call last):
  File "/Users/michael.collins/code/eliga/aldryn-redirects/tests/settings.py", line 41, in <module>
    run()
  File "/Users/michael.collins/code/eliga/aldryn-redirects/tests/settings.py", line 37, in run
    runner.cms('aldryn_redirects')
  File "/Users/michael.collins/code/eliga/aldryn-redirects/venv/lib/python3.9/site-packages/app_helper/runner.py", line 49, in cms
    return runner(argv)
  File "/Users/michael.collins/code/eliga/aldryn-redirects/venv/lib/python3.9/site-packages/app_helper/runner.py", line 109, in runner
    return main(argv)
  File "/Users/michael.collins/code/eliga/aldryn-redirects/venv/lib/python3.9/site-packages/app_helper/main.py", line 356, in main
    return core(args=args, application=application)
  File "/Users/michael.collins/code/eliga/aldryn-redirects/venv/lib/python3.9/site-packages/app_helper/main.py", line 283, in core
    _make_settings(args, application, settings, STATIC_ROOT, MEDIA_ROOT)
  File "/Users/michael.collins/code/eliga/aldryn-redirects/venv/lib/python3.9/site-packages/app_helper/utils.py", line 285, in _make_settings
    django.setup()
  File "/Users/michael.collins/code/eliga/aldryn-redirects/venv/lib/python3.9/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/Users/michael.collins/code/eliga/aldryn-redirects/venv/lib/python3.9/site-packages/django/apps/registry.py", line 122, in populate
    app_config.ready()
  File "/Users/michael.collins/code/eliga/aldryn-redirects/venv/lib/python3.9/site-packages/django/contrib/admin/apps.py", line 27, in ready
    self.module.autodiscover()
  File "/Users/michael.collins/code/eliga/aldryn-redirects/venv/lib/python3.9/site-packages/django/contrib/admin/__init__.py", line 24, in autodiscover
    autodiscover_modules('admin', register_to=site)
  File "/Users/michael.collins/code/eliga/aldryn-redirects/venv/lib/python3.9/site-packages/django/utils/module_loading.py", line 47, in autodiscover_modules
    import_module('%s.%s' % (app_config.name, module_to_search))
  File "/Users/michael.collins/.pyenv/versions/3.9.9/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/Users/michael.collins/code/eliga/aldryn-redirects/aldryn_redirects/admin.py", line 1, in <module>
    from tablib import Dataset
ModuleNotFoundError: No module named 'tablib'

Process finished with exit code 1
```

File: .github/workflows/test.yml (hunk 2 above)
Comment (Bernardvdv) [reply .github/workflows/test.yml:11]: Looks like all the major packages still support 3.7 so I guess it's fine

File: tests/requirements/requirements_base.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (adam-murray) [reply tests/requirements/requirements_base.txt:9]: Updated and tested on clean environment.

File: .github/workflows/test.yml (hunk 2 above)
Comment (adam-murray) [reply .github/workflows/test.yml:11]: Waiting on confirmation from the internal team on this.

Comment (adam-murray) [reply .github/workflows/test.yml:11]: Looks like we commented at almost exactly the same time! - I agree, but will see what the guys come back with.

**Review by michaeljcollinsuk — APPROVED** [review]

**Pull request merged**